### PR TITLE
refactor: streamline handles throughout the app

### PR DESCRIPTION
**Pull request by idoshamun**
Enforce the same rules on user and squad handles.

The rules:
* Transform to lowercase
* No forbidden handles
* Uses alphanumeric, underscore or dash
* No dups

The rules are enforced only when updating the handle or creating a new entity. We want to make sure users who update something else will not suddenly get an error.

WT-744 #done

### CODE REVIEW

File: src/entity/User.ts [hunk 1]
```diff
@@ -288,7 +292,13 @@ export const addNewUser = async (
         id: data.id,
         name: data.name,
         image: data.image ?? fallbackImages.avatar,
-        username: data.username,
+        username: data.username
```
Comment (idoshamun) [inline src/entity/User.ts:295]: Handle can be empty on creation (social login)

File: src/entity/User.ts [hunk 2]
```diff
@@ -149,20 +157,20 @@ const checkRequiredFields = (data: AddUserData): boolean => {
...
+  let query = entityManager
     .getRepository(User)
     .createQueryBuilder()
     .select('id')
-    .where('email = :email or username = :username', {
-      email,
-      username,
-    })
-    .getRawOne();
+    .where({ email: email.toLowerCase() });
```
[the remark this answers is not in the document]
Comment (omBratteng) [reply src/entity/User.ts:169]: Should we do a migration where we convert all emails to lowercase?

**Review by rebelchris — APPROVED** [review]
Nice work.

Should we not also retro fix the existing wrong handles?
We also need to update the disallowHandle table to include all next pages and `undefined` etc in regards to this task.

**Comment by idoshamun** [comment]
@rebelchris do we have a list of all the pages? That's a solid point!
As for the retro fix, once we apply this change I'll check how many users will be impacted and decide accordingly how to proceed

File: src/entity/User.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (idoshamun) [reply src/entity/User.ts:169]: I'd like to first asses the impact

**Comment by rebelchris** [comment]
@idoshamun Not sure if it's complete, but let's start by adding these:

- account (taken 🤦‍♂️)
- backoffice
- popup
- posts
- search
- sources
- squads
- tags
- 404
- bookmarks
- callback
- devcard
- discussed
- error
- filters
- history
- join
- my-feed
- notifications
- onboarding
- popular
- settings
- signup
- upvoted
- login (taken 🤦‍♂️)
- null (taken 🤦‍♂️)
- undefined (taken 🤦‍♂️)
- index

This this will cover for now?

Another [ticket](https://github.com/dailydotdev/daily/issues/855) that came to mind, did you add disallow for `-`? (fix before I think was just for mentions)
Maybe let's add a test for it.

Also related the above names [ticket](https://github.com/dailydotdev/daily/issues/833).
With this let's ensure you can't add `/` as well

Found another username related [ticket](https://github.com/dailydotdev/daily/issues/832).

**Comment by idoshamun** [comment]
@rebelchris I added the ones you mentioned to the disallow list already and addressed the other concerns. Please review again.
Another follow up task for the frontend will be updating the handle's regex. Please note that I separated it into two, one for social handles (github, etc) and one for our handle a bit more strict. Can you address that please?

**Review by rebelchris — APPROVED** [review]
Looks good to me :)
I'll add a task for FE handle updates

**Pull request merged**